ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt

```diff
@@ -92,17 +92,16 @@ internal fun CoroutineScope.attachForReadingDirectImpl(
         channel.writeSuspendSession {
             while (true) {
                 var rc = 0
+                val buffer = request(1)
+                if (buffer == null) {
+                    if (channel.isClosedForWrite) break
+                    channel.flush()
+                    tryAwait(1)
```

Comment:
Why don't we control timeout here anymore?

Reply:
We use timeout only for network operations

Reply:
btw it works in the same way in similar places

Reply:
yes, previous behavior was wrong